midend/interpreter.cpp

```diff
@@ -735,6 +747,18 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         auto li = l->to<SymbolicBool>();
         clone->expr = new IR::BoolLiteral(li->value);
         DoConstantFolding cf(refMap, typeMap);
+        if (auto cast = expression->to<IR::Cast>()) {
+            if (cast->destType->is<IR::Type_Bits>()) {
+                const IR::Constant* constant;
+                if (clone->expr) {
+                    constant = new IR::Constant(cast->destType, 1);
+                } else {
+                    constant = new IR::Constant(cast->destType, 0);
+                }
+                set(expression, new SymbolicInteger(constant->to<IR::Constant>()));
```

Comment:
Why is this cast necessary?

Reply:
Set type `bit<1>` for new constant. Wouldn't it be better to use `new IR :: Type_Bits` instead of `cast->destType`?